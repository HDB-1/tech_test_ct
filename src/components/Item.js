import React from 'react'

export default function Item(props) {
    return (
        <div>
            <p>showing Item component</p>
            <ul>
                {props.allInfo.map((allInfo, index) => <li key={index}>Name = {allInfo.name}</li>)}
            </ul>
            
        </div>
    )
}
