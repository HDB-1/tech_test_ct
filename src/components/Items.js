import React, { Component } from 'react'
import Item from './Item';

export default class Items extends Component {
    render() {
        return (
            <div>
                <p>showing items component</p>
                <Item allInfo={this.props.allInfo} />
            </div>
        )
    }
}
