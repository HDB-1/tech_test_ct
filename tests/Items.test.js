import React from 'react';
import { shallow } from 'enzyme';
import Items from '../src/components/Items';
import Item from '../src/components/Item';

describe("shallow Items", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<Items />)
    })
    it("should render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    })
    it("should contain an Item component", () => {
        expect(wrapper.containsMatchingElement(<Item />)).toEqual(true);
    })
})