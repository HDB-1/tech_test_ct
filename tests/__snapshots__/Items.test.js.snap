// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow Items should render correctly 1`] = `
<div>
  <p>
    showing items component
  </p>
  <Item />
</div>
`;
